docs/spec/nmt-wrapper.md

```diff
@@ -0,0 +1,84 @@
+# Namespaced Merkle Tree Wrapper
+## Abstract
+In Celestia, block transactions are grouped into identical-size shares and arranged in a `k` by `k` matrix, called [original data square](https://github.com/celestiaorg/celestia-app/blob/specs-staging/specs/src/specs/data_structures.md#arranging-available-data-into-shares).
+The rows and columns of the original data square is then extended using 2D Reed-Solomon coding scheme.
+The extended version is called extended data square, that is a `2k` by `2k` square consisting of `4` quadrants namely, `Q0`, `Q1`, `Q2`, and `Q3`.
+Figure 1 provides an illustration of a sample data square and its extended version.
+`Q0` corresponds to the original data square.
+`Q1` and `Q2` represent the horizontal and vertical extensions of `Q0`, respectively. 
+`Q3` is the extension of `Q2` in the horizontal direction.
+```markdown
+  k         k
+ -------   -------
+|        |        |
+|   Q0 → |   Q1   | k
+|   ↓    |        |
+-------  -------
+|        |        |
+|   Q2 → |   Q3   | k
+|        |        |
+ -------   -------
+```
```

Comment:
[non-blocking comment]

is there another place where we describe the square? perhaps we can link there as well, but

Reply:
Addressed https://github.com/celestiaorg/celestia-app/pull/1442/commits/d0519e7661a14c510b47de30a361ca7a23c55075.